app/redux/shared_actions/SetsActionCreators.js

```diff
@@ -81,7 +83,8 @@ export const endSet = (manuallyStarted=false, wasSanityCheck=false) => (dispatch
 
         dispatch({
             type: END_SET,
-            defaultMetric: defaultMetric
+            defaultMetric: defaultMetric,
+            sets: setIDs
```

Comment:
It looks like you're creating the model on end set. IMO the model should have a default, it should be wiped when the workout finishes / logging in / logging out, and it should only edit each set as they go.